mac

```diff
@@ -166,29 +166,62 @@ brew_install_or_upgrade 'hub'
 # shellcheck disable=SC2016
 append_to_file "$HOME/.zshrc" 'eval "$(hub alias -s)"'
 
+
+fancy_echo 'Checking on Node.js installation...'
+
 if ! brew_is_installed "node"; then
-  if ! command -v n > /dev/null; then
-    fancy_echo 'Installing n and latest Node.js and NPM...'
-    curl -L http://git.io/n-install | bash -s -- -y latest
+  if command -v nvm > /dev/null; then
+    if ! command -v n > /dev/null; then
+      fancy_echo 'Installing n and latest Node.js and NPM...'
+      curl -L http://git.io/n-install | bash -s -- -y latest
+    else
+      fancy_echo 'Updating n...'
+      n-update -y
+    fi
   else
-    fancy_echo 'Updating n...'
-    n-update -y
+    fancy_echo 'nvm detected.  Skipping...'
   fi
+else
+  brew_install_or_upgrade 'node'
 fi
 
-brew_install_or_upgrade 'python3'
+fancy_echo '...Finished Node.js installation checks.'
+
+
+fancy_echo 'Checking on Python installation...'
+
+if ! brew_is_installed "python3"; then
+  brew_install_or_upgrade 'pyenv'
+  brew_install_or_upgrade 'pyenv-virtualenv'
+  brew_install_or_upgrade 'pyenv-virtualenvwrapper'
+
+  # pyenv currently doesn't have a convenience version to use, e.g., "latest"
+  if ! pyenv versions | ag "3.5.1" > /dev/null; then
```

Comment:
I would prefer not to have to update the latest version manually every time a new version is released. RVM conveniently keeps the latest version in the GitHub repo, so I can [use curl to get the latest version](https://github.com/monfresh/laptop/blob/master/mac#L164). I was looking to see if Python provided a similar convenience, but I couldn't find it. The best I could come up with was to parse their OS X downloads page:
```bash
dirty_version=$(curl -s https://www.python.org/downloads/mac-osx/ | sed -n -e 's/^.*Latest Python 3 Release - Python //p')
latest_python_version=${dirty_version%</a></li>}
```
What do you think about using that to get the latest version?